app/controllers/vote_controller.rb

```diff
@@ -57,8 +57,8 @@ def refresh_vote_cache
     # Naming.refresh_vote_cache
     Observation.refresh_vote_cache
     flash_notice(:refresh_vote_cache.t)
-    redirect_with_query(controller: :observer,
-                        action: :list_rss_logs,
+    redirect_with_query(controller: :rss_logs,
+                        action: :index,
                         id: observation.id)
   end
```

Comment:
Oh my stars.  Have you looked at what this action actually does?  Or tries to do -- it would probably fail to do it in the age of the universe!  Fortunately, there is no link to this action.  Would you mind terribly ripping it out?  If we ever decide that we absolutely must recalculate the consensus for every single observation on the site, we need to rewrite Observation.refresh_vote_cache to take advantage of eager loading at very least.  I would recommend a mass-update, as well.  I think even if the calculation were trivial, simply running updates for every single observation in the database one at a time would be prohibitively slow.  I think if this has a prayer it will have to be rewritten to calculate a massive hash of all of the new consensus names and vote caches, and then change them all in one or a few gigantic mass updates.  (Then update the cached classification and text_name columns in observations table, too, but that part's already written.)

Reply:
By the way, the variable "observation" isn't even initialized the way this action is currently written, right?

Reply:
Let's rip it out. 💨

Reply:
Commented out, with additional comment about expensive query.